app/code/Magento/Eav/Model/Attribute/GroupRepository.php

```diff
@@ -117,16 +117,6 @@ public function save(\Magento\Eav\Api\Data\AttributeGroupInterface $group)
      */
     public function getList(\Magento\Framework\Api\SearchCriteriaInterface $searchCriteria)
     {
-        $attributeSetId = $this->retrieveAttributeSetIdFromSearchCriteria($searchCriteria);
```

Comment:
@maghamed Since you requested the attribute set to be optional in https://github.com/magento/magento2/issues/11936#issuecomment-342424893: are you fine with it being removed completely?

Reply:
hey @mzeis and @tzyganu 
I am fine with removing this check,
The only thing we need to add here is to mark method 
`protected function retrieveAttributeSetIdFromSearchCriteria(`
as `@deprecated` because we no need it any more, but because it's protected we can't just remove it